src/patterns/OcFormInput.vue

```diff
@@ -0,0 +1,191 @@
+<template>
+  <div>
+    <oc-text-input
+      v-bind="$_ocFormInput_inputBindings"
+      v-on="$listeners"
+      :has-error="!!errorMessage"
+      :has-warning="!!warningMessage"
+    />
+    <div class="oc-form-input-message" v-if="$_ocFormInput_showMessageLine">
+      <span v-if="!!warningMessage" class="oc-form-input-warning">{{ warningMessage }}</span>
+      <span v-if="!!errorMessage" class="oc-form-input-danger">{{ errorMessage }}</span>
+    </div>
+  </div>
+</template>
+
+<script>
+import OcTextInput from "../elements/OcTextInput"
+/**
+ * Form Inputs are used to allow users to provide text input when the expected
+ * input is short. Form Input has a range of options and supports several text
+ * formats including numbers. For longer input, use the form `Textarea` element.
+ *
+ * All properties will be handed through to the inner OcTextInput component. The purpose
+ * of this component is, to be able to show validation messages below the input, while
+ * utilizing all functionality from OcTextInput.
+ *
+ * ## Accessibility
+ * The attributes `placeholder` and `aria-label` have different functions. The first specifies a short hint describing the expected value of an input field/text area, or gives an example (e.g. email@example.com). `aria-label` provides the accessible name of the text input (e.g. "Your address", "Comment",...).
+ */
+export default {
+  name: "oc-form-input",
+  components: { OcTextInput },
+  status: "review",
+  release: "1.0.0",
+  inheritAttrs: false,
```

Comment:
FYI this will be enhanced in vue 3, so that `$attrs` will include classes and styles as well. This way you can pass down attributes, classes, styles and listeners into a child component. `v-bind` on the `oc-text-input` in line 4 passes $attrs and $props into the text input. Neat feature.

Reply:
`inheritAttrs: false` has the effect, that $attrs and $listeners are not applied to the root element of the template, but on whichever other element you decide instead.